plugins/@grouparoo/mongo/src/lib/table-import/util.ts

```diff
@@ -193,3 +184,80 @@ function getNestedValue(entity, fieldName) {
   }
   return entity;
 }
+
+export async function getFields(
+  connection: any,
+  tableName: string
+): Promise<Array<string>> {
+  const firstLevelFieldsResult = await connection.db
+    .collection(tableName)
+    .aggregate([
+      { $project: { keyValue: { $objectToArray: "$$ROOT" } } },
+      { $unwind: "$keyValue" },
+      { $group: { _id: null, keys: { $addToSet: "$keyValue.k" } } },
+    ])
+    .toArray();
+
+  if (firstLevelFieldsResult.length > 0) {
+    const allFields = await getAllFields(
+      connection,
+      tableName,
+      firstLevelFieldsResult[0].keys
+    );
+    return Array.from(allFields);
+  }
+  return [];
+}
+
+async function getAllFields(
+  connection: any,
+  tableName: string,
+  firstLevelFields: Array<string>
+): Promise<Set<string>> {
+  let allFields = new Set<string>();
+  const docs = await connection.db
+    .collection(tableName)
+    .aggregate([{ $sample: { size: 10 } }, { $project: { _id: 0 } }])
+    .toArray();
```

Comment:
Will this always return the same 10 records or is it random?  
Postgres and MySQL return the same 10 each time so either is OK... just curious.

Reply:
it obtains N random documents of the given collection.